File: medxc-gui/src/app/model/doctor.ts
import { Specialty } from './specialty';
import { Appointment } from './appointment';

export class Doctor {
    id: string;
    name: string;
    email: string;
    phoneNumber: string;
    active: boolean;
    specialties: Specialty[];
    appointments: Appointment[];

    constructor(id: string, name: string, email: string, phoneNumber: string, active: boolean,
                specialties: Specialty[], appointments: Appointment[]) {
        this.id = id;
        this.name = name;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.active = true;
        this.specialties = specialties;
        this.appointments = appointments;
    }
    getSpecialties() {
        return this.specialties;
    }
}

File: medxc-persistence/src/main/java/com/dxc/medxc/persistence/model/Doctor.java
/**
 *
 */
package com.dxc.medxc.persistence.model;

import java.util.List;

/**
 * @author yyayya
 */
public interface Doctor {

    /**
     * @return doctor's number
     */
    String getDocNum();

    /**
     * @return doctor's name
     */
    String getName();

    /**
     * @return doctor's email
     */
    String getEmail();

    /**
     * @return doctor's phone number
     */
    String getPhoneNumber();

    /**
     * @return list of doctors's appointments
     */
    List<Appointment> getAppointments();

    /**
     * @return specialties
     */
    List<DocSpec> getSpecialties();

    /**
     * @param specialties
     *            specialties of the doctor
     */
    void setDocSpec(final List<DocSpec> specialties);

}
